tests/backup/backup_test.go

```diff
@@ -86,6 +88,508 @@ var _ = Describe("{UserGroupManagement}", func() {
+				log.FailOnError(err, "Failed to switch to context to source cluster")
+
+				sourceClusterAppsContexts = make([]*scheduler.Context, 0)
+				for i := 0; i < Inst().GlobalScaleFactor; i++ {
+					taskName := fmt.Sprintf("%s-%d", taskNamePrefix, i)
+					appContexts = ScheduleApplications(taskName)
+					for _, appCtx := range appContexts {
+						appCtx.ReadinessTimeout = appReadinessTimeout
+						namespace := GetAppNamespace(appCtx, taskName)
+						// (sourceNamespaces, sourceClusterAppsContexts) will always correspoond
+						sourceNamespaces = append(sourceNamespaces, namespace)
+						sourceClusterAppsContexts = append(sourceClusterAppsContexts, appCtx)
+					}
+				}
+			})
+
+			Step("Validate applications", func() {
+				ValidateApplications(sourceClusterAppsContexts)
+
+				log.InfoD("Switching to default context")
+				err := SetClusterContext("")
+				log.FailOnError(err, "Failed to SetClusterContext to default cluster")
+			})
+
+			log.Warn("Waiting for 2 minutes, hoping for any Custom Resources to finish starting up.\nThis logic will have to be changed to polling in the future, as just randomly waiting, without verification can easily lead to errors. Make sure that any errors you're seeing is not due to this.")
+			time.Sleep(time.Minute * 2)
+
+		})
+
+		Step("Creating rules for backup", func() {
+			log.InfoD("Creating pre rule for deployed apps")
+			for i := 0; i < len(appList); i++ {
+				preRuleStatus, ruleName, err := Inst().Backup.CreateRuleForBackup(appList[i], orgID, "pre")
+				log.FailOnError(err, "Creating pre rule for deployed apps failed")
+				dash.VerifyFatal(preRuleStatus, true, "Verifying pre rule for backup")
+
+				if ruleName != "" {
+					preRuleNameList = append(preRuleNameList, ruleName)
+				}
+			}
+			log.InfoD("Creating post rule for deployed apps")
+			for i := 0; i < len(appList); i++ {
+				postRuleStatus, ruleName, err := Inst().Backup.CreateRuleForBackup(appList[i], orgID, "post")
+				log.FailOnError(err, "Creating post rule for deployed apps failed")
+				dash.VerifyFatal(postRuleStatus, true, "Verifying Post rule for backup")
+				if ruleName != "" {
+					postRuleNameList = append(postRuleNameList, ruleName)
+				}
+			}
+		})
+
+		Step("Creating bucket,backup location and cloud setting", func() {
+			log.InfoD("Creating bucket,backup location and cloud setting")
+			for _, provider := range providers {
+				cloudCredName = fmt.Sprintf("%s-%s-%v", "cred", provider, time.Now().Unix())
+				bkpLocationName = fmt.Sprintf("%s-%s-bl", provider, getGlobalBucketName(provider))
+				cloudCredUID = uuid.New()
+				backupLocationUID = uuid.New()
+				backupLocationMap[backupLocationUID] = bkpLocationName
+				CreateCloudCredential(provider, cloudCredName, cloudCredUID, orgID)
+				err := CreateBackupLocation(provider, bkpLocationName, backupLocationUID, cloudCredName, cloudCredUID, getGlobalBucketName(provider), orgID, "")
+				dash.VerifyFatal(err, nil, "Creating backup location")
```

Comment:
Mention the backup location name that is being created

Reply:
✅